### ISSUE TITLE
Is the bootstrap method implemented here a multiscale one?

### ISSUE THREAD

**Issue by EsqYu** [cdt15/lingam#95]
When using bootstrap to evaluate the causal structure given by LiNGAM, multiscale bootstrap is said to be better. Is the bootstrap method already prepared here a multiscale one?

**Comment by sshimizu2006** [comment]
Hi, multiscale bootstrap is not implemented in this package for LiNGAM. 

Though not specialized to LiNGAM, in general cases, R package for multi scale bootstrap is available: https://github.com/shimo-lab/scaleboot

**Comment by EsqYu** [comment]
Hi, thank you for your reply and introducing R package for multi scale bootstrap.
Can I somehow use this package to evaluate the result LiNGAM gives?

**Comment by sshimizu2006** [comment]
Yes. Basically, it would be something like computing bootstrap probabilities with different numbers of bootstrap resampling and then giving them to the R code.

**Comment by EsqYu** [comment]
Thanks for your reply.
Does "computing bootstrap probabilities with different numbers of bootstrap resampling" mean execute the bootstrap() method with different number of n_sampling like the following?
1)model.bootstrap(X, n_sampling=100)
2) model.bootstrap(X, n_sampling=200)
3) model.bootstrap(X, n_sampling=300)
4)give the results1〜3 to R code

**Comment by sshimizu2006** [comment]
Yes, something like that, though I don't know the details of the R package very much.

**Comment by EsqYu** [comment]
Thank you very much for your kind replies. I'll try using the package. By the way, I have one more question. I believe there is two options when selecting the model, 'pwling' or 'kernel'. How should I use these options differently?

**Comment by sshimizu2006** [comment]
Basically, pwling would be the first choice since it is faster to compute.

**Comment by EsqYu** [comment]
Okay. Then, What is the advantage of kernel version of it ?

**Comment by sshimizu2006** [comment]
kernel is a kind of nonparametric estimator of independence. pwling makes some distributional assumption like super-Gaussian distributions. See the details for these references: 
https://www.jmlr.org/papers/v14/hyvarinen13a.html for pwling, 
and https://www.jmlr.org/papers/v3/bach02a.html for kernel.

**Comment by EsqYu** [comment]
Thanks for your reply and sharing the URLs. I believed I should use the two methods separately depending on the features of the data used for input like skewness or kurtosis. Is this not that kind of thing?

**Comment by sshimizu2006** [comment]
Yeah, you can try both of them.

**Comment by EsqYu** [comment]
Thank you. I tried multiscale bootstrapping using the R package, and I got some results, but in the example, they use log-likelihood values as input. Is it appropriate to use bootstrap probabilities as input?". 
I'm also wondering how many times I should do bootstrap before using this package.
